### DISCUSSION TITLE
Using 100 instances of express-rate-limit

### DISCUSSION THREAD

**Discussion by MohamedKarrab** [nfriedly/express-rate-limit#448] [Q&A]
Is there a performance issue with using 100 instances of express-rate-limit? I have a lot of routes and I want to assign each of them different rate limiting (windows size, key, and handler). Or is there a better way to do this?

**Comment by nfriedly** [comment]
If any given request will only hit one instance, then that's probably fine. If you use any store aside from the built-in memory store, you'll need to assign a unique `prefix` for each instance so that they don't interfere with each other. (Although I suppose unique keys would also solve this.)

I'd recommend aiming for shorter window sizes in general, which will help keep the overhead down.
